model/src/main/java/com/javahelp/model/userEntityTest/UserPassword.java

```diff
@@ -0,0 +1,99 @@
+package com.javahelp.model.userEntityTest;
+
+import java.util.Base64;
+
+public class UserPassword {
+    /**
+     * A class for hashing a user's password
+     * --- Attributes ---
+     * salt - a current salt of a user
+     * hash - the current and hashed password of the user
+     * user - the user object
+     */
+
+    private byte[] salt;
+
+    private byte[] hash;
+
+    private User user;
```

Comment:
In my opinion, the `UserPassword` class should not contain a reference to the `User` the password belongs to. The purpose of the class should more be to bundle together a user's salt and hash, and also to disconnect this information from the `User` object, so this information is not needlessly loaded into memory.